src/coreclr/jit/gentree.h

```diff
@@ -9680,6 +9683,47 @@ inline bool GenTree::IsMaskAllBitsSet() const
     return false;
 }
 
+inline bool GenTree::IsAllTrue() const
+{
+#ifdef TARGET_ARM64
+    return IsMaskAllBitsSet() || IsVectorAllBitsSet();
+#endif
+    return false;
+}
+
+inline bool GenTree::IsAllFalse() const
+{
+#ifdef TARGET_ARM64
+    return IsMaskAllBitsZero() || IsVectorZero();
+#endif
+    return false;
+}
+
+inline bool GenTree::IsMaskAllBitsZero() const
+{
+#ifdef TARGET_ARM64
+    static_assert_no_msg(AreContiguous(NI_Sve_CreateFalseMaskByte, NI_Sve_CreateFalseMaskDouble,
+                                       NI_Sve_CreateFalseMaskInt16, NI_Sve_CreateFalseMaskInt32,
+                                       NI_Sve_CreateFalseMaskInt64, NI_Sve_CreateFalseMaskSByte,
+                                       NI_Sve_CreateFalseMaskSingle, NI_Sve_CreateFalseMaskUInt16,
+                                       NI_Sve_CreateFalseMaskUInt32, NI_Sve_CreateFalseMaskUInt64));
+
+    if (OperIsHWIntrinsic())
+    {
+        NamedIntrinsic id = AsHWIntrinsic()->GetHWIntrinsicId();
+        if (id == NI_Sve_ConvertMaskToVector)
+        {
+            GenTree* op1 = AsHWIntrinsic()->Op(1);
+            assert(op1->OperIsHWIntrinsic());
+            id = op1->AsHWIntrinsic()->GetHWIntrinsicId();
+        }
+        return ((id >= NI_Sve_CreateFalseMaskByte) && (id <= NI_Sve_CreateFalseMaskUInt64));
+    }
```

Comment:
This seems like a case that'd benefit from it being a `GenTreeMaskCon` so we can have a normalized and cheap view of the concept here.

Reply:
Not something to handle here, just noting more generally as the concept also applies to xarch and people may have the wrong expectations